src/components/scrollBar/index.js

```diff
@@ -126,40 +146,61 @@ class ScrollBar extends BaseComponent {
 
   onLayout = ({nativeEvent}) => {
     this.containerWidth = nativeEvent.layout.width;
-    
+
     // 1 - for race condition, in case onContentSizeChange() is called before
     // 0 - for containerWidth change, when onContentSizeChange() is called first
     this.setState({gradientOpacity: new Animated.Value(this.scrollContentWidth > this.containerWidth ? 1 : 0)});
-  }
+  };
+
+  onItemLayout = ({layout, index}) => {
+    this.itemsLayouts[index] = layout;
+  };
 
   renderScrollable() {
-    const {useList, forwardedRef} = this.props;
+    const {useList, forwardedRef, children} = this.props;
     const Component = useList ? FlatList : ScrollView;
 
     return (
       <Component
+        scrollEventThrottle={100}
         {...this.getThemeProps()}
-        ref={forwardedRef}
+        ref={r => {
+          this.scrollbar = r;
+          if (_.isFunction(forwardedRef)) {
+            forwardedRef(r);
+          } else if (_.isObject(forwardedRef)) {
+            forwardedRef.current = r;
+          }
+        }}
         horizontal
         showsHorizontalScrollIndicator={false}
         onScroll={this.onScroll}
         onContentSizeChange={this.onContentSizeChange}
-      />
+      >
+        {children &&
+          React.Children.map(children, (child, index) => {
+            return (
+              <Item onLayout={this.onItemLayout} index={index}>
```

Comment:
Why do we need to wrap the child with another View? Can't we just alter its props and return it?

Reply:
I wanted to avoid cloning, similar to what we did in the Carousel. 
Also, cloning might be problematic when I'll use it in TabController

Reply:
I meant something like: `
React.Children.map(children, (child, index) => {
      child.props.onLayout = this.onItemLayout;
      child.props.index = index;
      return child;
});
`
Meaning alter the child's props but return it without a wrapper view. Can it work?

Reply:
you can do that, that's why there's the cloneElement api 
We'll have to actually clone the children in order to do that..

Reply:
ok